code/obj/ToSplit/barber_shop.dm

```diff
@@ -123,200 +185,137 @@
 	anchored = 1
 	desc = "Barber poles historically were signage used to convey that the barber would perform services such as blood letting and other medical procedures, with the red representing blood, and the white representing the bandaging. In America, long after the time when blood-letting was offered, a third colour was added to bring it in line with the colours of their national flag. This one is in space."
 
-///////////////////////////////////////////////////
-//////Hair Dye Bottle Code					///////
-///////////////////////////////////////////////////
-/obj/item/dye_bottle/attack(mob/living/carbon/human/M as mob, mob/user as mob)
-	if(!ishuman(M))	return
-	if(user.zone_sel.selecting != "head" || user.a_intent != "help")
-		..()
-		return
-	if(src.empty)
+
+/obj/item/proc/dye_hair(mob/living/carbon/human/M as mob, mob/user as mob, obj/item/dye_bottle/bottle as obj)
```

Comment:
Why is this definted on /obj/item ????

Reply:
Not sure why the dye bottle's defined there, I cant think of any other items that'd be suitable.